### ISSUE TITLE
HTML reports for bst test

### ISSUE THREAD

**Issue by dmarvp** [bespoken/bst#541]
**Is your feature request related to a problem? Please describe.**
Users want to be able to share their tests results as html as well.

**Describe the solution you'd like**
An argument in bst test that will generate html reports if desired.

**Additional context**
The following library can be used: https://github.com/Hargne/jest-html-reporter
Also, in the testing json this could be added to make it work (but the idea would be to have it on demand):

```json
{
  "jest": {
    "reporters": [
        "default",
        [
          "./node_modules/jest-html-reporter",
          {
             "includeFailureMsg": true,
               "pageTitle": "Bespoken Test Report"
          }
        ]
      ],
    "silent": false,
    "testMatch": ["**/test/*.yml", "**/tests/*.yml", "**/*.e2e.yml", "**/*.spec.yml"]
  }
}
```

**Comment by jkelvie** [comment]
I found this other HTML reporter for Jest. Looks to be a bit prettier. We should check this out as well, and see which one we like better:
https://github.com/dkelosky/jest-stare

**Comment by jperata** [comment]
the parameter can be "reporterType", it would automatically fill the jest.reporters field internally if set to "HTML".
It can be set to "none",  or not include it to keep the default behavior. And we can create a different report type (csv or pdf) if required.

**Comment by jkelvie** [comment]
Here is another optionI found - this also looks rather nice:
https://github.com/Hazyzh/jest-html-reporters

**Comment by jperata** [comment]
This is are demos of the reporters using "guess the price" 
the jest-html-reporter
<img width="1672" alt="screen shot 2019-01-30 at 4 05 29 pm" src="https://user-images.githubusercontent.com/8781604/52016098-b1a98900-24b1-11e9-8e6d-2b59eb063232.png">

the jest-stare
<img width="542" alt="screen shot 2019-01-30 at 4 10 21 pm" src="https://user-images.githubusercontent.com/8781604/52016201-e61d4500-24b1-11e9-9e58-60bd73221683.png">

<img width="1662" alt="screen shot 2019-01-30 at 4 50 17 pm" src="https://user-images.githubusercontent.com/8781604/52016162-cede5780-24b1-11e9-8494-a1beeebdf8b3.png">
<img width="1214" alt="screen shot 2019-01-30 at 4 50 38 pm" src="https://user-images.githubusercontent.com/8781604/52016170-d271de80-24b1-11e9-99a7-78eee5785636.png">

the jest-html-reporters

<img width="1674" alt="screen shot 2019-01-30 at 4 08 27 pm" src="https://user-images.githubusercontent.com/8781604/52016221-f6cdbb00-24b1-11e9-8335-61300b12f2ee.png">
<img width="1675" alt="screen shot 2019-01-30 at 4 08 36 pm" src="https://user-images.githubusercontent.com/8781604/52016239-0220e680-24b2-11e9-9148-1b17b70996ec.png">
<img width="1623" alt="screen shot 2019-01-30 at 4 08 46 pm" src="https://user-images.githubusercontent.com/8781604/52016248-064d0400-24b2-11e9-8c83-f24a162b1ab7.png">

I will go with the last one.

**Comment by jkelvie** [comment]
Last one looks awesome. What do the errors look like on it?

**Comment by jperata** [comment]
This is a run with a couple of errors
<img width="1670" alt="screen shot 2019-01-30 at 5 38 53 pm" src="https://user-images.githubusercontent.com/8781604/52017657-0c44e400-24b6-11e9-872f-53baa96ad072.png">
<img width="1637" alt="screen shot 2019-01-30 at 5 39 01 pm" src="https://user-images.githubusercontent.com/8781604/52017658-0c44e400-24b6-11e9-8010-80a81ac0b91b.png">
<img width="1598" alt="screen shot 2019-01-30 at 5 39 18 pm" src="https://user-images.githubusercontent.com/8781604/52017659-0cdd7a80-24b6-11e9-8861-2397a60de963.png">

If you click on the info button
<img width="1612" alt="screen shot 2019-01-30 at 5 39 30 pm" src="https://user-images.githubusercontent.com/8781604/52017660-0cdd7a80-24b6-11e9-80b8-9c5f743a6bd9.png">

**Comment by jkelvie** [comment]
I love it.

**Comment by dmarvp** [comment]
Can something be done with all those NaN:NaN:NaN?

**Comment by jperata** [comment]
Can be tested in version 2.1.20 (current beta)

**Comment by MKen-li** [comment]
Moving back to dev

**Comment by jperata** [comment]
Can be tested in version 2.1.25 (current beta)

**Comment by jkelvie** [comment]
Closed by #566
